nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/CommonClassLoaderServiceImpl.java

```diff
@@ -189,87 +178,57 @@ private static String urlsToClassPath(Stream<URL> urls) {
         return urls.map(FileUtils::toFile).map(File::getAbsolutePath).collect(Collectors.joining(File.pathSeparator));
     }
 
-    private List<File> findDerbyClient() {
-        final String DERBY_HOME_PROP = "AS_DERBY_INSTALL";
-        StartupContext startupContext = env.getStartupContext();
-        Properties arguments = null;
+    private List<File> findDerbyJars(File installDir) {
+        Path derbyHome = getDerbyDir(installDir);
+        LOG.log(CONFIG, "Using derby home: {0}", derbyHome);
 
-        if (startupContext != null) {
-            arguments = startupContext.getArguments();
+        final File derbyLib = derbyHome.resolve("lib").toFile();
+        if (!derbyLib.exists()) {
+            LOG.info(KernelLoggerInfo.cantFindDerby);
+            return Collections.emptyList();
         }
 
-        String derbyHome = null;
-
-        if (arguments != null) {
-            derbyHome = arguments.getProperty(DERBY_HOME_PROP,
-                System.getProperty(DERBY_HOME_PROP));
-        }
+        return Arrays
+            .asList(derbyLib.listFiles((dir, name) -> name.endsWith(".jar") && !name.startsWith("derbyLocale_")));
```

Comment:
Is it worth to introduce a constant for `.jar`? It's also used in `JarFileFilter` class

Reply:
Not sure, because we are messing with classloaders. Any new/removed dependency can change order of initialization which can break something.